benchmarks/cugraph/standalone/cugraph_bulk_sampling.py

```diff
@@ -0,0 +1,618 @@
+                    replication_factor,
+                    {
+                        'node': node_offsets[node_type]
+                    },
+                    meta=cudf.DataFrame({'node':cudf.Series(dtype='int64')})
+                ).persist()
+
+            gc.collect()
+    
+    node_labels_df = dask_cudf.concat(
+        list(node_labels.values())
+    ).persist()
+    
+    del node_labels
+    gc.collect()
+
+    return all_edges_df, node_labels_df, node_offsets_replicated, edge_offsets, total_num_nodes
+    
+
+def benchmark_cugraph_bulk_sampling(dataset, output_path, seed, batch_size, seeds_per_call, fanout, reverse_edges=True, dataset_dir='.', replication_factor=1, num_labels=256, labeled_percentage=0.001):
+    """
+    Entry point for the benchmark.
+
+    Parameters
+    ----------
+    dataset: str
+        The dataset to sample.  Can be rmat_{scale}_{edgefactor}, or the name of an ogb dataset.
+    output_path: str
+        The output path, where samples and metadata will be stored.
+    seed: int
+        The random seed.
+    batch_size: int
+        The batch size (number of input seeds in a single sampling batch).
+    seeds_per_call: int
+        The number of input seeds in a single sampling call.
+    fanout: list[int]
+        The fanout.
+    reverse_edges: bool
+        Whether to reverse edges when constructing the graph.
+    dataset_dir: str
+        The directory where datasets are stored (only for ogb datasets)
+    replication_factor: int
+        The number of times to replicate the dataset.
+    num_labels: int
+        The number of random labels to generate (only for rmat datasets)
+    labeled_percentage: float
+        The percentage of the data that is labeled (only for rmat datasets)
+        Defaults to 0.001 to match papers100M
+    """
+    if dataset[0:5] == 'rmat':
+        dataset = dataset.split('_')
+        scale = int(dataset[1])
+        edgefactor = int(dataset[2])
+
+        dask_edgelist_df = generate_edgelist_rmat(
+            scale=scale, edgefactor=edgefactor, seed=seed, unweighted=True, mg=True,
+        )
+        dask_edgelist_df = dask_edgelist_df.astype("int64")
+        dask_edgelist_df = dask_edgelist_df.reset_index(drop=True)
+
+
+        dask_edgelist_df = renumber_ddf(dask_edgelist_df).persist()
+        dask_edgelist_df = symmetrize_ddf(dask_edgelist_df).persist()
+        dask_edgelist_df['etp'] = cupy.int32(0) # doesn't matter what the value is, really
+        
+        # generator = np.random.default_rng(seed=seed)
+        num_labeled_nodes = int(2**(scale+1) * labeled_percentage)
+        label_df = pd.DataFrame({
+            'node': np.arange(num_labeled_nodes),
+            # 'label': generator.integers(0, num_labels - 1, num_labeled_nodes).astype('float32')
+        })
+        
+        dask_label_df = ddf.from_pandas(label_df)
+        del label_df
+        gc.collect()
+
+        dask_label_df = dask_cudf.from_dask_dataframe(dask_label_df)
```

Comment:
Minor nitpick of moving this to a function so that we can use it at other places.

Reply:
resolved